source/krill/multi-user/config-file-provider.rst

```diff
@@ -0,0 +1,201 @@
+that are relevant for that user.
+
+When a user enters their username and password into the web user interface a 
+hash of the password is computed and sent with the username to the Krill server.
+
+The Krill server will verify that the user logging in provided a correct
+password and has the ``LOGIN`` permission. On success Krill will respond with a
+token which the web user interface should send on subsequent requests to
+authenticate itself with Krill. The web user interface will keep a copy of this
+token in browser local storage until the user logs out or is timed out due to
+inactivity.
+
+.. tip:: The actual user password is **NEVER** stored on either the Krill server
+         nor the client browser and is **NEVER** sent by the client browser to
+         the Krill server. Only password *hashes* are stored and transmitted.
+
+.. warning:: Do **NOT** serve the Krill web user interface over unencrypted HTTP.
+             While the password is never transmitted, the authentication token
+             that the user is subsequently issued is subject to interception
+             by malicious parties if sent unencrypted from the Krill server to
+             the web user interface. Note that this is equally true when using
+             any credential to authenticate with Krill, whether secret token
+             or password hash or when Krill is configured to interact with an
+             OpenID Connect provider.
+
+.. _limitations:
+
+Known limitations
+-----------------
+
+Config File Users are easy to define and give you complete control over who
+has access to your Krill instance and what level of access is granted. However,
+Krill is not a complete user management system and so there are some things to
+remember when using Config File Users:
+
+- Krill has no feature for requiring a user to change their password on first
+  login. As such, by issuing users with passwords you become responsible for
+  delivering the new password to them securely.
+
+- OpenID Connect providers often have support for one-time passwords (OTP)
+  or other secondary lines of defence to protect an account than just a
+  username and password. Krill does not have this capability.
+
+- Krill has no feature for generating cryptographically strong passwords. You
+  are responsible for choosing sufficiently strong passwords for your users.
+
+- Usernames, password hashes and user attributes are sensitive information. By
+  adding them to your ``krill.conf`` file you become responsible for protecting
+  them.
+
+- If you lose you ``krill.conf`` file you will also lose the password hashes
+  and will have to reset your users passwords unless you have a (**secure**)
+  copy elsewhere.
+
+- If a user forgets their password you will need to issue them with a new one.
+  Krill does not offer a forgotten password or password reset feature.
+
+- Adding or changing users requires a restart of Krill. There is no support in
```

Comment:
The way this is phrased a user can be left with the impression this can take a long time. Can we can any indiction how long it takes to start/restart Krill?

Reply:
In my experience it's pretty instantaneous. @timbru ?